pjmedia/src/pjmedia-videodev/ffmpeg_dev.c

```diff
@@ -449,8 +449,8 @@ static pj_status_t ffmpeg_factory_refresh(pjmedia_vid_dev_factory *f)
                 continue;
 
             for(i = 0; i < ctx->nb_streams; i++) {
-                if (ctx->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO) {
-                    codec = ctx->streams[i]->codec;
+                if (ctx->streams[i]->codecpar->codec_type == AVMEDIA_TYPE_VIDEO) {
+                    codec = ctx->streams[i]->codecpar;
```

Comment:
Perhaps it is better to put some version checks here? Something like 
https://github.com/pjsip/pjproject/blob/b34497b3cfcf388005bb9a203db86310508e796d/pjmedia/src/pjmedia-videodev/ffmpeg_dev.c#L180-L186

Not sure since which version `AVStream.codec` is deprecated, perhaps we can just follow [this](https://git.jerryxiao.cc/Android/scrcpy/commit/b98eb7d0fafd347704a6826d5996c59dd12ee301?files=app) (re: ffmpeg vs libav versions, the higher version should be okay as it seems to be removed much later after deprecated).

Reply:
Yeah, this is a more suite way to slove this, after I check the commit history of [https://git.ffmpeg.org/ffmpeg.git](https://git.ffmpeg.org/ffmpeg.git), I found in this commit , libavformat remove these releate code as below:
![Screenshot from 2025-05-09 17-35-27](https://github.com/user-attachments/assets/9b26901e-eebc-4f14-aef9-f13f97648049)
And the releate libavformat verion as below:
![Screenshot from 2025-05-09 17-35-03](https://github.com/user-attachments/assets/eba21c2d-197f-4f3a-9787-f228e5d1eb79)

So I add a line code with ``LIBAVFORMAT_VER_AT_LEAST(59,0)`` here.

Reply:
Perhaps the check version should refer to deprecation rather than removal, as some libraries can be built without the deprecated API. However, this is fine to me.

Reply:
I thought when these code is marked with deprecation, the compile process still can build. so I use the removal version to compare, just try to keep compile success.